collect_app/src/test/java/org/odk/collect/android/widgets/WidgetFactoryTest.java

```diff
@@ -0,0 +1,345 @@
+package org.odk.collect.android.widgets;
+
+import android.app.Activity;
+
+import org.javarosa.core.model.Constants;
+import org.javarosa.core.model.QuestionDef;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.support.MockFormEntryPromptBuilder;
+import org.odk.collect.android.support.RobolectricHelpers;
+import org.odk.collect.android.support.TestScreenContextActivity;
+import org.odk.collect.android.utilities.ActivityAvailability;
+import org.odk.collect.android.utilities.QuestionMediaManager;
+import org.odk.collect.android.widgets.items.LabelWidget;
+import org.odk.collect.android.widgets.items.LikertWidget;
+import org.odk.collect.android.widgets.items.ListMultiWidget;
+import org.odk.collect.android.widgets.items.ListWidget;
+import org.odk.collect.android.widgets.items.SelectMultiImageMapWidget;
+import org.odk.collect.android.widgets.items.SelectMultiMinimalWidget;
+import org.odk.collect.android.widgets.items.SelectMultiWidget;
+import org.odk.collect.android.widgets.items.SelectOneImageMapWidget;
+import org.odk.collect.android.widgets.items.SelectOneMinimalWidget;
+import org.odk.collect.android.widgets.items.SelectOneWidget;
+import org.odk.collect.android.widgets.support.FakeQuestionMediaManager;
+import org.odk.collect.android.widgets.support.FakeWaitingForDataRegistry;
+import org.odk.collect.android.widgets.utilities.WaitingForDataRegistry;
+import org.robolectric.RobolectricTestRunner;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(RobolectricTestRunner.class)
+public class WidgetFactoryTest {
+    private WidgetFactory widgetFactory;
+
+    @Before
+    public void setup() {
+        Activity activity = RobolectricHelpers.buildThemedActivity(TestScreenContextActivity.class).get();
+        WaitingForDataRegistry waitingForDataRegistry = new FakeWaitingForDataRegistry();
+        QuestionMediaManager questionMediaManager = new FakeQuestionMediaManager();
+        ActivityAvailability activityAvailability = mock(ActivityAvailability.class);
+
+        widgetFactory = new WidgetFactory(activity,
+                false,
+                false,
+                waitingForDataRegistry,
+                questionMediaManager,
+                null,
+                null,
+                activityAvailability,
+                null);
+    }
+
+    @Test
+    public void testCreatingSelectOneMinimalWidgetWithFastExternalChoices() {
```

Comment:
This is a pretty bad case with two data sources. External itemsets "win" because they're first? Since this test is not about the two data source weirdness, I would strongly suggest removing it and just having either an external select or a search() select.

Reply:
I removed this tests since it was redundant.